src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2898,9 +2912,11 @@ void smt2_convt::convert_plus(const plus_exprt &expr)
 
       out << ")";
     }
-    else if(expr.type().id()==ID_rational)
+    else if(expr.type().id()==ID_rational ||
+            expr.type().id()==ID_integer ||
+            expr.type().id()==ID_real)
```

Comment:
Why would `ID_integer` or `ID_real` have two operands?

Reply:
It's in the branch "if(expr.operands().size()==2)", and otherwise, make_binary is called.

However, this is of course unnecessary, as SMT2 + on both integers and reals is multi-ary.